internals/workloads/workloads.go

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a Copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workloads
+
+import (
+	"errors"
+	"fmt"
+	"maps"
+	"reflect"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/pebble/internals/osutil"
+	"github.com/canonical/pebble/internals/plan"
+)
+
+var _ plan.SectionExtension = (*PlanExtension)(nil)
+
+type PlanExtension struct{}
```

Comment:
Consider moving everything related to `PlanExtension` into a different file in this package.